pkg/collector/corechecks/embed/jmx/runner.go

```diff
@@ -30,10 +29,6 @@ func (r *runner) initRunner() {
 func (r *runner) startRunner() error {
 
 	lifecycleMgmt := true
```

Comment:
This is a good question. I guess the only reason to keep it around is if we want to make it configurable via a user-facing (maybe hidden) config option.

Reply:
There is a reason actually: when the restart logic triggers, it uses the same function but passing `false` [1]. That's because the thread running the restart logic will continue running and we don't want to start another one.

[1] https://github.com/DataDog/datadog-agent/pull/5309/files#diff-7389c2caae3e69d2d88fd7cf12dc8792R152